extensions/amp-script/0.1/test/integration/test-amp-script.js

```diff
@@ -46,15 +48,36 @@ describe.configure().skipSinglePass().run('amp-script', function() {
     it('should say "hello world"', function*() {
       yield poll('<amp-script> to be hydrated',
           () => element.classList.contains('i-amphtml-hydrated'));
+      const impl = element.implementation_;
 
       // Give event listeners in hydration a moment to attach.
       yield browser.wait(100);
 
+      sandbox.stub(impl.userActivation_, 'isActive').callsFake(() => true);
       browser.click('button#hello');
       yield poll('Hello World!', () => {
         const h1 = doc.querySelector('h1');
         return h1 && h1.textContent == 'Hello World!';
       });
     });
+
+    it('should terminate without gesture', function*() {
+      yield poll('<amp-script> to be hydrated',
+          () => element.classList.contains('i-amphtml-hydrated'));
+      const impl = element.implementation_;
+
+      // Give event listeners in hydration a moment to attach.
+      yield browser.wait(100);
+
+      sandbox.stub(impl.userActivation_, 'isActive').callsFake(() => false);
+      let terminated = false;
+      sandbox.stub(impl.workerDom_, 'terminate').callsFake(() => {
+        terminated = true;
+      });
+      browser.click('button#hello');
```

Comment:
Ditto. Instead of checking for `terminate()` invocation, maybe we can wait for X ms and ensure that the text didn't mutate (X as small as possible).